client/state/data-layer/wpcom/sites/hosting/php-version.js

```diff
@@ -0,0 +1,115 @@
+/**
+ * Internal dependencies
+ */
+import { http } from 'state/data-layer/wpcom-http/actions';
+import { dispatchRequest } from 'state/data-layer/wpcom-http/utils';
+import { registerHandlers } from 'state/data-layer/handler-registry';
+import {
+	HOSTING_PHP_VERSION_REQUEST,
+	HOSTING_PHP_VERSION_SET_REQUEST,
+	HOSTING_PHP_VERSION_SET,
+} from 'state/action-types';
+import { errorNotice, successNotice } from 'state/notices/actions';
+import { translate } from 'i18n-calypso';
+import { composeAnalytics, recordGoogleEvent, recordTracksEvent } from 'state/analytics/actions';
+
+const updateNoticeId = 'hosting-php-version';
+
+const getPhpVersion = action =>
+	http(
+		{
+			method: 'GET',
+			path: `/sites/${ action.siteId }/hosting/php-version`,
+			apiNamespace: 'wpcom/v2',
+			body: {},
+		},
+		action
+	);
+
+const getPhpVersionSuccess = ( action, version ) => {
+	return {
+		type: HOSTING_PHP_VERSION_SET,
+		siteId: action.siteId,
+		version,
+	};
+};
+
+const updatePhpVersion = action =>
+	http(
+		{
+			method: 'POST',
+			path: `/sites/${ action.siteId }/hosting/php-version`,
+			apiNamespace: 'wpcom/v2',
+			body: {
+				version: action.version,
+			},
+		},
+		action
+	);
+
+export const hostingPhpVersionUpdateTracking = ( version, result ) =>
+	composeAnalytics(
+		recordGoogleEvent(
+			'Hosting Configuration',
+			'Clicked "Update PHP Version" Button in PHP Version box',
+			`PHP Version Update ${ result }`,
+			version
+		),
+		recordTracksEvent( 'calypso_hosting_configuration_php_version_update', {
+			result,
+			version,
+		} )
+	);
+
+const updatePhpVersionSuccess = action => {
+	return [
+		hostingPhpVersionUpdateTracking( action.version, true ),
+		{
+			type: HOSTING_PHP_VERSION_SET,
+			siteId: action.siteId,
+			version: action.version,
+		},
+		successNotice(
+			translate( 'PHP version successfully set to %(version)s.', {
+				args: {
+					version: action.version,
+				},
+			} ),
+			{
+				id: updateNoticeId,
+				showDismiss: false,
+				duration: 5000,
+			}
+		),
+	];
+};
+
+const updatePhpVersionError = action => {
+	return [
+		hostingPhpVersionUpdateTracking( action.version, false ),
+		errorNotice( translate( 'Failed to set PHP version.' ), {
+			id: updateNoticeId,
+		} ),
+		{
+			type: HOSTING_PHP_VERSION_SET,
```

Comment:
Fire this, for the loading state to be off. :)

Reply:
Is not the loading state linked to the set request? The need of firing this action sounds more like a bug. Anyway, if we move to just 2 actions and update client state optimistically, this shouldn't be needed.

Reply:
Forgot to remove, good catch. I don't like the optimistic update. I'd rather stick with the 3 actions.

Reply:
> Forgot to remove, good catch. I don't like the optimistic update. I'd rather stick with the 3 actions.

Agreed, especially since this is hopefully a low-frequency change :)